src/docs/extensions.md

```diff
@@ -2,16 +2,27 @@
 title: Extensions
 ---
 
-# Extensions
+# Extensions and plugins
 
-Theia is composed of extensions. An extension is an npm package that exposes
-any number of DI modules (`ContainerModule`) that contribute to the creation of
-the DI container.
+Eclipse Theia is designed in a very modular and extensible way. It supports three ways of being extended/adapted to your specific requirements. These are complementary and are targeted at different use cases. You can pick the best option and even mix them within the same project.
+In the following, we will give you a quick overview of the available extension mechanisms and provide more details in the sections below.
 
-An extension is consumed by adding a dependency to the
-npm-package in the `package.json` of the application.
-Extensions can be installed/uninstalled at
-runtime, which will trigger a recompilation and restart.
+- **VS Code extensions**: Simple to write, installable at runtime, compatible with VS Code, limited to the VS Code extension API, some use cases not possible due to API restrictions => Used for adding features to existing tools. Note that you can also use existing VS Code extensions within Theia, too.
```

Comment:
Perhaps bullet points would be clearer than a run-on sentence with pros and cons mixed together?

Reply:
Had that before, looks pretty weird. I would prefer to keep it as is.